lisp/forge-topic.el

```diff
@@ -403,26 +430,18 @@ identifier."
         (magit-insert-section (note)
           (magit-insert-heading "Note")
           (insert (forge--fontify-markdown note) "\n\n")))
-      (dolist (post (cons topic (oref topic posts)))
-        (with-slots (author created body) post
-          (magit-insert-section section (post post)
-            (oset section heading-highlight-face
-                  'magit-diff-hunk-heading-highlight)
-            (let ((heading
-                   (format-spec
-                    forge-post-heading-format
-                    `((?a . ,(propertize (or author "(ghost)")
-                                         'font-lock-face 'forge-post-author))
-                      (?c . ,(propertize created 'font-lock-face 'forge-post-date))
-                      (?C . ,(propertize (apply #'format "%s %s ago"
-                                                (magit--age
-                                                 (float-time
-                                                  (date-to-time created))))
-                                         'font-lock-face 'forge-post-date))))))
-              (add-face-text-property 0 (length heading)
-                                      'magit-diff-hunk-heading t heading)
-              (magit-insert-heading heading))
-            (insert (forge--fontify-markdown body) "\n\n"))))
+      ;; insert topic description
+      (with-slots (author created body) topic
+        (magit-insert-section section (post topic)
+          (oset section heading-highlight-face 'forge-post-heading-highlight)
+          (forge--insert-section author created body 'forge-post-heading)))
+      ;; insert posts related to the topic
+      (dolist (post posts)
+        (unless (when (forge-pullreq-p topic) (oref post diff-p))
```

Comment:
In such cases that are about the returned value, as opposed to some side-effect, you should always use `and` instead of `when`. This `unless` on the other hand is "correct" because that *is* about the side-effect of its body.